In [5]:
import warnings as wn
wn.filterwarnings("ignore")

# Read in the CSV data
import pandas as pd
school_data=r"C:\Data Analytics Course\Capstone Projects\Capstone 2\schools_complete.csv"
student_data=r"C:\Data Analytics Course\Capstone Projects\Capstone 2\students_complete.csv"

# Read School and Student Data File and store into Pandas DataFrames
school_data = pd.read_csv(school_data)
student_data = pd.read_csv(student_data)

# Combine the data into a single dataset.  
school_data_complete = pd.merge(student_data, school_data, how="left", on=["school_name", "school_name"])
school_data_complete.head()

Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  School ID      type  size   budget  
0             66          79          0  District  2917  1910635  
1             94          61          0  District  2917  1910635  
2             90          60          0  District  2917  1910635  
3             67          58          0  District  2917  1910635  
4             97          84          0  District  2917  1910635

DISTRICT SUMMARY

In [9]:
# Calculate the total number of unique schools
total_schools = school_data_complete["school_name"].nunique() 
total_schools

15

In [11]:
# Calculate the total number of students
total_students = school_data_complete["student_name"].count()
total_students

39170

In [15]:
# Calculate the total budget
total_budget = school_data["budget"].sum()
total_budget

24649428

In [17]:
# Calculate the average (mean) math score
average_math_score = school_data_complete["math_score"].mean()
average_math_score

78.98537145774827

In [19]:
# Calculate the average (mean) reading score
average_reading_score = school_data_complete["reading_score"].mean()
average_reading_score

81.87784018381414

In [23]:
# Use the following to calculate the percentage of students who passed math (math scores greather than or equal to 70)
passing_math_count = school_data_complete[(school_data_complete["math_score"] >= 70)].count()["student_name"]
passing_math_percentage = passing_math_count / float(total_students) * 100
passing_math_percentage

74.9808526933878

In [29]:
# Calculate the percentage of students who passed reading (hint: look at how the math percentage was calculated)  
passing_reading_count = school_data_complete[(school_data_complete["reading_score"] >= 70)].count()["student_name"]
passing_reading_percentage = passing_reading_count / float(total_students) * 100
passing_reading_percentage

85.80546336482001

In [33]:
# Use the following to calculate the percentage of students that passed math and reading
passing_math_reading_count = school_data_complete[
    (school_data_complete["math_score"] >= 70) & (school_data_complete["reading_score"] >= 70)
].count()["student_name"]
overall_passing_rate = passing_math_reading_count /  float(total_students) * 100
overall_passing_rate

65.17232575950983

In [39]:
# Create a high-level snapshot of the district's key metrics in a DataFrame
district_summary = pd.DataFrame({
    "Total Schools": [total_schools],
    "Total Students": [total_students],
    "Total Budget": [total_budget],
    "Average Math Score": [average_math_score],
    "Average Reading Score": [average_reading_score],
    "% Passing Math": [passing_math_percentage],
    "% Passing Reading": [passing_reading_percentage],
    "% Overall Passing": [overall_passing_rate]
})

# Formatting
district_summary["Total Students"] = district_summary["Total Students"].map("{:,}".format)
district_summary["Total Budget"] = district_summary["Total Budget"].map("${:,.2f}".format)

# Display the DataFrame
district_summary

Total Schools Total Students    Total Budget  Average Math Score  \
0             15         39,170  $24,649,428.00           78.985371   

   Average Reading Score  % Passing Math  % Passing Reading  % Overall Passing  
0               81.87784       74.980853          85.805463          65.172326

SCHOOL SUMMARY

In [43]:
# Group data by each school
school_group = school_data_complete.groupby(["school_name"])

# Use the code provided to select all of the school types
school_type = school_data.set_index("school_name")["type"]

In [45]:
# Calculate the total student count per school
total_students_per_school = school_group["student_name"].count()
total_budget_per_school = school_group["budget"].mean()
per_student_budget = total_budget_per_school / total_students_per_school
average_math_score_per_school = school_group["math_score"].mean()
average_reading_score_per_school = school_group["reading_score"].mean()

In [47]:
# % Passing per school
passing_math_per_school = school_group.apply(lambda x: (x["math_score"] >= 70).mean()) * 100
passing_reading_per_school = school_group.apply(lambda x: (x["reading_score"] >= 70).mean()) * 100
overall_passing_per_school = school_group.apply(lambda x: ((x["math_score"] >= 70) & (x["reading_score"] >= 70)).mean()) * 100

In [199]:
# Create School Summary DataFrame
per_school_summary = pd.DataFrame({
    "School Type": school_type,
    "Total Students": total_students_per_school,
    "Total School Budget": total_budget_per_school,
    "Per Student Budget": per_student_budget,
    "Average Math Score": average_math_score_per_school,
    "Average Reading Score": average_reading_score_per_school,
    "% Passing Math": passing_math_per_school,
    "% Passing Reading": passing_reading_per_school,
    "% Overall Passing": overall_passing_per_school
})

per_school_summary

School Type  Total Students  Total School Budget  \
school_name                                                              
Bailey High School       District            4976            3124928.0   
Cabrera High School       Charter            1858            1081356.0   
Figueroa High School     District            2949            1884411.0   
Ford High School         District            2739            1763916.0   
Griffin High School       Charter            1468             917500.0   
Hernandez High School    District            4635            3022020.0   
Holden High School        Charter             427             248087.0   
Huang High School        District            2917            1910635.0   
Johnson High School      District            4761            3094650.0   
Pena High School          Charter             962             585858.0   
Rodriguez High School    District            3999            2547363.0   
Shelton High School       Charter            1761            1056600.0   
Thomas High School        Charter            1635            1043130.0   
Wilson High School        Charter            2283            1319574.0   
Wright High School        Charter            1800            1049400.0   

                       Per Student Budget  Average Math Score  \
school_name                                                     
Bailey High School                  628.0           77.048432   
Cabrera High School                 582.0           83.061895   
Figueroa High School                639.0           76.711767   
Ford High School                    644.0           77.102592   
Griffin High School                 625.0           83.351499   
Hernandez High School               652.0           77.289752   
Holden High School                  581.0           83.803279   
Huang High School                   655.0           76.629414   
Johnson High School                 650.0           77.072464   
Pena High School                    609.0           83.839917   
Rodriguez High School               637.0           76.842711   
Shelton High School                 600.0           83.359455   
Thomas High School                  638.0           83.418349   
Wilson High School                  578.0           83.274201   
Wright High School                  583.0           83.682222   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Bailey High School                 81.033963       66.680064   
Cabrera High School                83.975780       94.133477   
Figueroa High School               81.158020       65.988471   
Ford High School                   80.746258       68.309602   
Griffin High School                83.816757       93.392371   
Hernandez High School              80.934412       66.752967   
Holden High School                 83.814988       92.505855   
Huang High School                  81.182722       65.683922   
Johnson High School                80.966394       66.057551   
Pena High School                   84.044699       94.594595   
Rodriguez High School              80.744686       66.366592   
Shelton High School                83.725724       93.867121   
Thomas High School                 83.848930       93.272171   
Wilson High School                 83.989488       93.867718   
Wright High School                 83.955000       93.333333   

                       % Passing Reading  % Overall Passing  
school_name                                                  
Bailey High School             81.933280          54.642283  
Cabrera High School            97.039828          91.334769  
Figueroa High School           80.739234          53.204476  
Ford High School               79.299014          54.289887  
Griffin High School            97.138965          90.599455  
Hernandez High School          80.862999          53.527508  
Holden High School             96.252927          89.227166  
Huang High

## Highest-Performing Schools (by % Overall Passing)

In [53]:
# Sort the schools by `% Overall Passing` in descending order and display the top 5 rows.
top_schools = per_school_summary.sort_values("% Overall Passing", ascending=False)
top_schools.head(5)

School Type  Total Students Total School Budget  \
school_name                                                           
Cabrera High School     Charter            1858       $1,081,356.00   
Thomas High School      Charter            1635       $1,043,130.00   
Griffin High School     Charter            1468         $917,500.00   
Wilson High School      Charter            2283       $1,319,574.00   
Pena High School        Charter             962         $585,858.00   

                    Per Student Budget  Average Math Score  \
school_name                                                  
Cabrera High School            $582.00           83.061895   
Thomas High School             $638.00           83.418349   
Griffin High School            $625.00           83.351499   
Wilson High School             $578.00           83.274201   
Pena High School               $609.00           83.839917   

                     Average Reading Score  % Passing Math  % Passing Reading  \
school_name                                                                     
Cabrera High School              83.975780       94.133477          97.039828   
Thomas High School               83.848930       93.272171          97.308869   
Griffin High School              83.816757       93.392371          97.138965   
Wilson High School               83.989488       93.867718          96.539641   
Pena High School                 84.044699       94.594595          95.945946   

                     % Overall Passing  
school_name                             
Cabrera High School          91.334769  
Thomas High School           90.948012  
Griffin High School          90.599455  
Wilson High School           90.582567  
Pena High School             90.540541

## Bottom Performing Schools (By % Overall Passing)

In [55]:
# Sort the schools by `% Overall Passing` in ascending order and display the top 5 rows.
bottom_schools = per_school_summary.sort_values("% Overall Passing", ascending=True)
bottom_schools.head(5)

School Type  Total Students Total School Budget  \
school_name                                                             
Rodriguez High School    District            3999       $2,547,363.00   
Figueroa High School     District            2949       $1,884,411.00   
Huang High School        District            2917       $1,910,635.00   
Hernandez High School    District            4635       $3,022,020.00   
Johnson High School      District            4761       $3,094,650.00   

                      Per Student Budget  Average Math Score  \
school_name                                                    
Rodriguez High School            $637.00           76.842711   
Figueroa High School             $639.00           76.711767   
Huang High School                $655.00           76.629414   
Hernandez High School            $652.00           77.289752   
Johnson High School              $650.00           77.072464   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Rodriguez High School              80.744686       66.366592   
Figueroa High School               81.158020       65.988471   
Huang High School                  81.182722       65.683922   
Hernandez High School              80.934412       66.752967   
Johnson High School                80.966394       66.057551   

                       % Passing Reading  % Overall Passing  
school_name                                                  
Rodriguez High School          80.220055          52.988247  
Figueroa High School           80.739234          53.204476  
Huang High School              81.316421          53.513884  
Hernandez High School          80.862999          53.527508  
Johnson High School            81.222432          53.539172

## Math Scores by Grade

In [79]:
# Use the code provided to separate the data by grade
ninth_graders = school_data_complete[(school_data_complete["grade"] == "9th")]
tenth_graders = school_data_complete[(school_data_complete["grade"] == "10th")]
eleventh_graders = school_data_complete[(school_data_complete["grade"] == "11th")]
twelfth_graders = school_data_complete[(school_data_complete["grade"] == "12th")]

# Group by `school_name` and take the mean of the `math_score` column for each grade
ninth_grade_math_scores = ninth_graders.groupby("school_name")["math_score"].mean()
tenth_grade_math_scores = tenth_graders.groupby("school_name")["math_score"].mean()
eleventh_grade_math_scores = eleventh_graders.groupby("school_name")["math_score"].mean()
twelfth_grade_math_scores = twelfth_graders.groupby("school_name")["math_score"].mean()

# Combine each of the scores above into a single DataFrame called `math_scores_by_grade`
math_scores_by_grade = pd.DataFrame({
    "9th Grade": ninth_grade_math_scores,
    "10th Grade": tenth_grade_math_scores,
    "11th Grade": eleventh_grade_math_scores,
    "12th Grade": twelfth_grade_math_scores
})

# Minor data wrangling: remove the index name
math_scores_by_grade.index.name = None

# Display the DataFrame
math_scores_by_grade

9th Grade  10th Grade  11th Grade  12th Grade
Bailey High School     77.083676   76.996772   77.515588   76.492218
Cabrera High School    83.094697   83.154506   82.765560   83.277487
Figueroa High School   76.403037   76.539974   76.884344   77.151369
Ford High School       77.361345   77.672316   76.918058   76.179963
Griffin High School    82.044010   84.229064   83.842105   83.356164
Hernandez High School  77.438495   77.337408   77.136029   77.186567
Holden High School     83.787402   83.429825   85.000000   82.855422
Huang High School      77.027251   75.908735   76.446602   77.225641
Johnson High School    77.187857   76.691117   77.491653   76.863248
Pena High School       83.625455   83.372000   84.328125   84.121547
Rodriguez High School  76.859966   76.612500   76.395626   77.690748
Shelton High School    83.420755   82.917411   83.383495   83.778976
Thomas High School     83.590022   83.087886   83.498795   83.497041
Wilson High School     83.085578   83.724422   83.195326   83.035794
Wright High School     83.264706   84.010288   83.836782   83.644986

## Reading Score by Grade 

In [91]:
# Use the code provided to separate the data by grade
ninth_graders = school_data_complete[(school_data_complete["grade"] == "9th")]
tenth_graders = school_data_complete[(school_data_complete["grade"] == "10th")]
eleventh_graders = school_data_complete[(school_data_complete["grade"] == "11th")]
twelfth_graders = school_data_complete[(school_data_complete["grade"] == "12th")]

# Group by `school_name` and take the mean of the `reading_score` column for each grade
ninth_grade_reading_scores = ninth_graders.groupby("school_name")["reading_score"].mean()
tenth_grade_reading_scores = tenth_graders.groupby("school_name")["reading_score"].mean()
eleventh_grade_reading_scores = eleventh_graders.groupby("school_name")["reading_score"].mean()
twelfth_grade_reading_scores = twelfth_graders.groupby("school_name")["reading_score"].mean()

# Combine each of the scores above into a single DataFrame called `reading_scores_by_grade`
reading_scores_by_grade = pd.DataFrame({
    "9th": ninth_grade_reading_scores,
    "10th": tenth_grade_reading_scores,
    "11th": eleventh_grade_reading_scores,
    "12th": twelfth_grade_reading_scores
})

# Minor data wrangling: reorder the columns and remove the index name
reading_scores_by_grade = reading_scores_by_grade[["9th", "10th", "11th", "12th"]]
reading_scores_by_grade.index.name = None

# Display the DataFrame
reading_scores_by_grade

9th       10th       11th       12th
Bailey High School     81.303155  80.907183  80.945643  80.912451
Cabrera High School    83.676136  84.253219  83.788382  84.287958
Figueroa High School   81.198598  81.408912  80.640339  81.384863
Ford High School       80.632653  81.262712  80.403642  80.662338
Griffin High School    83.369193  83.706897  84.288089  84.013699
Hernandez High School  80.866860  80.660147  81.396140  80.857143
Holden High School     83.677165  83.324561  83.815534  84.698795
Huang High School      81.290284  81.512386  81.417476  80.305983
Johnson High School    81.260714  80.773431  80.616027  81.227564
Pena High School       83.807273  83.612000  84.335938  84.591160
Rodriguez High School  80.993127  80.629808  80.864811  80.376426
Shelton High School    84.122642  83.441964  84.373786  82.781671
Thomas High School     83.728850  84.254157  83.585542  83.831361
Wilson High School     83.939778  84.021452  83.764608  84.317673
Wright High School     83.833333  83.812757  84.156322  84.073171

## Scores by School Spending

In [143]:
# Define the spending bins and labels
spending_bins = [0, 585, 630, 645, 680]
spending_labels = ["<$585", "$585-630", "$630-645", "$645-680"]

In [203]:
# Create a copy of the school summary since it has the "Per Student Budget" 
school_spending_df = per_school_summary.copy()

In [207]:
# Use `pd.cut` to categorize spending based on the bins.
school_spending_df["Spending Ranges (Per Student)"] = pd.cut(per_school_summary["Per Student Budget"], bins=spending_bins, labels=spending_labels)

# Formatting
school_spending_df["Total School Budget"] = school_spending_df["Total School Budget"].map("${:,.2f}".format)
school_spending_df["Per Student Budget"] = school_spending_df["Per Student Budget"].map("${:,.2f}".format)
school_spending_df.head()


school_spending_df

School Type  Total Students Total School Budget  \
school_name                                                             
Bailey High School       District            4976       $3,124,928.00   
Cabrera High School       Charter            1858       $1,081,356.00   
Figueroa High School     District            2949       $1,884,411.00   
Ford High School         District            2739       $1,763,916.00   
Griffin High School       Charter            1468         $917,500.00   
Hernandez High School    District            4635       $3,022,020.00   
Holden High School        Charter             427         $248,087.00   
Huang High School        District            2917       $1,910,635.00   
Johnson High School      District            4761       $3,094,650.00   
Pena High School          Charter             962         $585,858.00   
Rodriguez High School    District            3999       $2,547,363.00   
Shelton High School       Charter            1761       $1,056,600.00   
Thomas High School        Charter            1635       $1,043,130.00   
Wilson High School        Charter            2283       $1,319,574.00   
Wright High School        Charter            1800       $1,049,400.00   

                      Per Student Budget  Average Math Score  \
school_name                                                    
Bailey High School               $628.00           77.048432   
Cabrera High School              $582.00           83.061895   
Figueroa High School             $639.00           76.711767   
Ford High School                 $644.00           77.102592   
Griffin High School              $625.00           83.351499   
Hernandez High School            $652.00           77.289752   
Holden High School               $581.00           83.803279   
Huang High School                $655.00           76.629414   
Johnson High School              $650.00           77.072464   
Pena High School                 $609.00           83.839917   
Rodriguez High School            $637.00           76.842711   
Shelton High School              $600.00           83.359455   
Thomas High School               $638.00           83.418349   
Wilson High School               $578.00           83.274201   
Wright High School               $583.00           83.682222   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Bailey High School                 81.033963       66.680064   
Cabrera High School                83.975780       94.133477   
Figueroa High School               81.158020       65.988471   
Ford High School                   80.746258       68.309602   
Griffin High School                83.816757       93.392371   
Hernandez High School              80.934412       66.752967   
Holden High School                 83.814988       92.505855   
Huang High School                  81.182722       65.683922   
Johnson High School                80.966394       66.057551   
Pena High School                   84.044699       94.594595   
Rodriguez High School              80.744686       66.366592   
Shelton High School                83.725724       93.867121   
Thomas High School                 83.848930       93.272171   
Wilson High School                 83.989488       93.867718   
Wright High School                 83.955000       93.333333   

                       % Passing Reading  % Overall Passing  \
school_name                                                   
Bailey High School             81.933280          54.642283   
Cabrera High School            97.039828          91.334769   
Figueroa High School           80.739234          53.204476   
Ford High School               79.299014          54.289887   
Griffin High School            97.138965          90.599455   
Hernandez High School          80.862999          53.527508   
Holden High School             96.252927          89.227166   
Huang High School              81.3

In [209]:
#  Calculate averages for the desired columns. 
spending_math_scores = school_spending_df.groupby(["Spending Ranges (Per Student)"])["Average Math Score"].mean()
spending_reading_scores = school_spending_df.groupby(["Spending Ranges (Per Student)"])["Average Reading Score"].mean()
spending_passing_math = school_spending_df.groupby(["Spending Ranges (Per Student)"])["% Passing Math"].mean()
spending_passing_reading = school_spending_df.groupby(["Spending Ranges (Per Student)"])["% Passing Reading"].mean()
overall_passing_spending = school_spending_df.groupby(["Spending Ranges (Per Student)"])["% Overall Passing"].mean()

In [213]:
# Assemble into DataFrame

spending_summary = pd.DataFrame({
    "Average Math Score": spending_math_scores,
    "Average Reading Score": spending_reading_scores,
    "% Passing Math": spending_passing_math,
    "% Passing Reading": spending_passing_reading,
    "% Overall Passing": overall_passing_spending
})

# Display results
spending_summary

Average Math Score  Average Reading Score  \
Spending Ranges (Per Student)                                              
<$585                                   83.455399              83.933814   
$585-630                                81.899826              83.155286   
$630-645                                78.518855              81.624473   
$645-680                                76.997210              81.027843   

                               % Passing Math  % Passing Reading  \
Spending Ranges (Per Student)                                      
<$585                               93.460096          96.610877   
$585-630                            87.133538          92.718205   
$630-645                            73.484209          84.391793   
$645-680                            66.164813          81.133951   

                               % Overall Passing  
Spending Ranges (Per Student)                     
<$585                                  90.369459  
$585-630                               81.418596  
$630-645                               62.857656  
$645-680                               53.526855

## Scores by School Size

In [215]:
# Establish the bins.
size_bins = [0, 1000, 2000, 5000]
labels = ["Small (<1000)", "Medium (1000-2000)", "Large (2000-5000)"]

In [219]:
# Categorize the spending based on the bins
# Use `pd.cut` on the "Total Students" column of the `per_school_summary` DataFrame.

per_school_summary["School Size"] = per_school_summary["School Size"] = pd.cut(per_school_summary["Total Students"], bins=size_bins, labels=labels)

per_school_summary

School Type  Total Students  Total School Budget  \
school_name                                                              
Bailey High School       District            4976            3124928.0   
Cabrera High School       Charter            1858            1081356.0   
Figueroa High School     District            2949            1884411.0   
Ford High School         District            2739            1763916.0   
Griffin High School       Charter            1468             917500.0   
Hernandez High School    District            4635            3022020.0   
Holden High School        Charter             427             248087.0   
Huang High School        District            2917            1910635.0   
Johnson High School      District            4761            3094650.0   
Pena High School          Charter             962             585858.0   
Rodriguez High School    District            3999            2547363.0   
Shelton High School       Charter            1761            1056600.0   
Thomas High School        Charter            1635            1043130.0   
Wilson High School        Charter            2283            1319574.0   
Wright High School        Charter            1800            1049400.0   

                       Per Student Budget  Average Math Score  \
school_name                                                     
Bailey High School                  628.0           77.048432   
Cabrera High School                 582.0           83.061895   
Figueroa High School                639.0           76.711767   
Ford High School                    644.0           77.102592   
Griffin High School                 625.0           83.351499   
Hernandez High School               652.0           77.289752   
Holden High School                  581.0           83.803279   
Huang High School                   655.0           76.629414   
Johnson High School                 650.0           77.072464   
Pena High School                    609.0           83.839917   
Rodriguez High School               637.0           76.842711   
Shelton High School                 600.0           83.359455   
Thomas High School                  638.0           83.418349   
Wilson High School                  578.0           83.274201   
Wright High School                  583.0           83.682222   

                       Average Reading Score  % Passing Math  \
school_name                                                    
Bailey High School                 81.033963       66.680064   
Cabrera High School                83.975780       94.133477   
Figueroa High School               81.158020       65.988471   
Ford High School                   80.746258       68.309602   
Griffin High School                83.816757       93.392371   
Hernandez High School              80.934412       66.752967   
Holden High School                 83.814988       92.505855   
Huang High School                  81.182722       65.683922   
Johnson High School                80.966394       66.057551   
Pena High School                   84.044699       94.594595   
Rodriguez High School              80.744686       66.366592   
Shelton High School                83.725724       93.867121   
Thomas High School                 83.848930       93.272171   
Wilson High School                 83.989488       93.867718   
Wright High School                 83.955000       93.333333   

                       % Passing Reading  % Overall Passing  \
school_name                                                   
Bailey High School             81.933280          54.642283   
Cabrera High School            97.039828          91.334769   
Figueroa High School           80.739234          53.204476   
Ford High School               79.299014          54.289887   
Griffin High School            97.138965          90.599455   
Hernandez High School          80.862999          53.527508   
Holden High School             96.252927          89.227166   
H

In [221]:
# Calculate averages for the desired columns. 
size_math_scores = per_school_summary.groupby(["School Size"])["Average Math Score"].mean()
size_reading_scores = per_school_summary.groupby(["School Size"])["Average Reading Score"].mean()
size_passing_math = per_school_summary.groupby(["School Size"])["% Passing Math"].mean()
size_passing_reading = per_school_summary.groupby(["School Size"])["% Passing Reading"].mean()
size_overall_passing = per_school_summary.groupby(["School Size"])["% Overall Passing"].mean()

In [223]:
# Create a DataFrame called `size_summary` that breaks down school performance based on school size (small, medium, or large).
# Use the scores above to create a new DataFrame called `size_summary`
size_summary = pd.DataFrame({
    "Average Math Score": size_math_scores,
    "Average Reading Score": size_reading_scores,
    "% Passing Math": size_passing_math,
    "% Passing Reading": size_passing_reading,
    "% Overall Passing": size_overall_passing
})

# Display the size summary DataFrame
size_summary 


Average Math Score  Average Reading Score  % Passing Math  \
School Size                                                                     
Small (<1000)                83.821598              83.929843       93.550225   
Medium (1000-2000)           83.374684              83.864438       93.599695   
Large (2000-5000)            77.746417              81.344493       69.963361   

                    % Passing Reading  % Overall Passing  
School Size                                               
Small (<1000)               96.099437          89.883853  
Medium (1000-2000)          96.790680          90.621535  
Large (2000-5000)           82.766634          58.286003

## Scores by School Type

In [225]:
# Group the per_school_summary DataFrame by "School Type" and average the results.
average_math_score_by_type = per_school_summary.groupby(["School Type"])["Average Math Score"].mean()
average_reading_score_by_type = per_school_summary.groupby(["School Type"])["Average Reading Score"].mean()
average_percent_passing_math_by_type = per_school_summary.groupby(["School Type"])["% Passing Math"].mean()
average_percent_passing_reading_by_type = per_school_summary.groupby(["School Type"])["% Passing Reading"].mean()
average_percent_overall_passing_by_type = per_school_summary.groupby(["School Type"])["% Overall Passing"].mean()

In [227]:
# Assemble the new data by type into a DataFrame called `type_summary`
type_summary = pd.DataFrame({
    "Average Math Score": average_math_score_by_type,
    "Average Reading Score": average_reading_score_by_type,
    "% Passing Math": average_percent_passing_math_by_type,
    "% Passing Reading": average_percent_passing_reading_by_type,
    "% Overall Passing": average_percent_overall_passing_by_type
})

# Display results
type_summary

Average Math Score  Average Reading Score  % Passing Math  \
School Type                                                              
Charter               83.473852              83.896421       93.620830   
District              76.956733              80.966636       66.548453   

             % Passing Reading  % Overall Passing  
School Type                                        
Charter              96.586489          90.432244  
District             80.799062          53.672208


Written Report: Academic Performance Analysis
District-Wide Observations
The analysis of academic performance metrics across various schools in the district reveals significant insights into the educational landscape.

Average Scores:

Average Math Score: The average math score for charter schools is 83.47, significantly outperforming district schools, which have an average score of 76.96.
Average Reading Score: Similarly, charter schools achieve an average reading score of 83.90, while district schools average 80.97.
Percentage Passing:

Passing Rates: Charter schools boast a passing rate of 93.62% in math and 96.59% in reading. In stark contrast, district schools show much lower passing rates of 66.55% for math and 80.80% for reading.
Overall Passing Rate: The overall passing rate for charter schools is 90.43%, compared to just 53.67% for district schools, highlighting a critical gap in student achievement.
Budget Analysis:

Total School Budget: Charter schools tend to operate with a lower per-student budget, averaging $582 compared to district schools, which average around $628 to $655 per student. Despite the apparent financial constraints, charter schools deliver superior academic performance.
Trend Analysis
The following two observable trends emerge from the data analysis:

Spending per Student vs. Performance:

A notable trend is that higher per-student spending does not always correlate with higher academic performance. For example, several district schools with higher budgets (e.g., Bailey High School at $628 per student) have lower passing rates and average scores than charter schools with less funding (e.g., Cabrera High School at $582 per student). This trend suggests that factors such as educational strategy, teacher quality, and student engagement may play a more significant role in academic success than financial resources alone.
School Size vs. Performance:

The analysis indicates that school size significantly influences academic performance. Smaller schools (less than 1000 students) exhibit higher average scores and passing rates compared to larger schools (2000-5000 students). For instance, small schools like Holden High School have an average math score of 83.80, while large schools such as Hernandez High School average only 77.29. This trend may point to smaller student-teacher ratios, allowing for more personalized instruction and support, fostering a more conducive learning environment.
Conclusion
The analysis reveals a stark contrast between charter and district schools in terms of academic performance and budget allocations. While charter schools achieve higher average scores and passing rates with lower per-student spending, district schools struggle despite often higher budgets. Moreover, the size of the schools also plays a critical role, with smaller schools generally outperforming larger ones.

These findings underscore the need for educational policymakers to consider not only financial investments but also the effectiveness of educational strategies and the importance of school size in fostering student success. Further research and targeted interventions may be necessary to address the disparities observed in district schools, ensuring equitable educational opportunities for all students.